sql/parser/sql.y

```diff
@@ -2774,14 +2774,6 @@ func_expr_windowless:
 // Special expressions that are considered to be functions.
 func_expr_common_subexpr:
   COLLATION FOR '(' a_expr ')' { unimplemented() }
-| CURRENT_DATE
```

Comment:
Huh, did I miss that discussion? I think this is a mistake. And it's not like this a burden to support. FYI, your "modern approach" information is incorrect. We need to support `CURRENT_{TIMESTAMP,DATE}` without parentheses.